src/tools/perf/ucp_tests.cc

```diff
@@ -440,6 +460,35 @@ class ucp_perf_test_runner {
         return UCS_PTR_IS_PTR(data) ? UCS_OK : UCS_PTR_STATUS(data);
     }
 
+    ucs_status_t UCS_F_ALWAYS_INLINE
+    recv_stream(ucp_ep_h ep, void *buf, unsigned length, ucp_datatype_t datatype,
+                uint8_t sn)
+    {
+        ssize_t  total = 0;
+        void    *rreq;
+        size_t   rlength;
+        ssize_t  rlength_s;
+
+        do {
+            rreq = ucp_stream_recv_nb(ep, buf, length - total, datatype,
+                                      (ucp_stream_recv_callback_t)ucs_empty_function,
+                                      &rlength, 0);
+            if (ucs_unlikely(UCS_PTR_IS_ERR(rreq))) {
+                return UCS_PTR_STATUS(rreq);
+            } else if (UCS_PTR_STATUS(rreq) == UCS_OK) {
```

Comment:
`if (ucs_likely(rreq == NULL))` should be first check, also to serve as an example for users